x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/filter/ShardBulkInferenceActionFilter.java

```diff
@@ -0,0 +1,534 @@
+                            var acc = inferenceResults.get(request.index);
+                            if (result instanceof ErrorChunkedInferenceResults error) {
+                                acc.addFailure(
+                                    new ElasticsearchException(
+                                        "Exception when running inference id [{}] on field [{}]",
+                                        error.getException(),
+                                        inferenceProvider.model.getInferenceEntityId(),
+                                        request.field
+                                    )
+                                );
+                            } else {
+                                acc.addOrUpdateResponse(
+                                    new FieldInferenceResponse(
+                                        request.field(),
+                                        request.input(),
+                                        request.inputOrder(),
+                                        request.isOriginalFieldInput(),
+                                        inferenceProvider.model,
+                                        result
+                                    )
+                                );
+                            }
+                        }
+                    } finally {
+                        onFinish();
+                    }
+                }
+
+                @Override
+                public void onFailure(Exception exc) {
+                    try {
+                        for (int i = 0; i < requests.size(); i++) {
+                            var request = requests.get(i);
+                            addInferenceResponseFailure(
+                                request.index,
+                                new ElasticsearchException(
+                                    "Exception when running inference id [{}] on field [{}]",
+                                    exc,
+                                    inferenceProvider.model.getInferenceEntityId(),
+                                    request.field
+                                )
+                            );
+                        }
+                    } finally {
+                        onFinish();
+                    }
+                }
+
+                private void onFinish() {
+                    if (nextBatch.isEmpty()) {
+                        onFinish.close();
+                    } else {
+                        executeShardBulkInferenceAsync(inferenceId, inferenceProvider, nextBatch, onFinish);
+                    }
+                }
+            };
+            inferenceProvider.service()
+                .chunkedInfer(
+                    inferenceProvider.model(),
+                    null,
+                    inputs,
+                    Map.of(),
+                    InputType.INGEST,
+                    new ChunkingOptions(null, null),
+                    TimeValue.MAX_VALUE,
```

Comment:
Ah, I was reading our docs, and apparently the bulk index timeout is specifically mentioning only a subset of things we respect with the user provided timeout.

We should be very clear that inference doesn't respect your timeouts. 

I foresee us having to add another timeout, or allowing the bulk index timeout to be supplied here.

Reply:
that is correct, the timeout only applies to very specific actions and not inference itself. We'll probably need to add specific timeouts in the future 👍